libraries/Microsoft.Bot.Builder.Core/Adapters/TestAdapter.cs

```diff
@@ -74,7 +76,13 @@ public new TestAdapter Use(IMiddleware middleware)
             List<ResourceResponse> responses = new List<ResourceResponse>();
 
             foreach (var activity in activities)
-            {                
+            {
+                if (String.IsNullOrEmpty(activity.Id))
+                    activity.Id = Guid.NewGuid().ToString("n");
```

Comment:
.... but.... what if we run out of GUIDs?? :) 

(Silly Q - do we want this to be a guid? For the Test Adapter, I think "int" would be better with a static counter. Certainly, it would make debugging easier.